test/uk/gov/hmrc/eusubsidycompliance/connectors/EisConnectorSpec.scala

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.eusubsidycompliance.connectors
+
+import com.github.tomakehurst.wiremock.client.WireMock._
+import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.test.Helpers.running
+import uk.gov.hmrc.eusubsidycompliance.models.json.digital.EisBadResponseException
+import uk.gov.hmrc.eusubsidycompliance.models.types.EORI
+import uk.gov.hmrc.eusubsidycompliance.test.Fixtures._
+import uk.gov.hmrc.eusubsidycompliance.test.util.WiremockSupport
+import uk.gov.hmrc.http.{HeaderCarrier, UpstreamErrorResponse}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class EisConnectorSpec extends AnyWordSpecLike with Matchers with WiremockSupport with ScalaFutures
+  with IntegrationPatience {
+
+  private val RetrieveUndertakingPath = "/scp/retrieveundertaking/v1"
+  private val CreateUndertakingPath = "/scp/createundertaking/v1"
+  private val RetrieveSubsidyPath = "/scp/getundertakingtransactions/v1"
+
+  implicit private val hc: HeaderCarrier = HeaderCarrier()
+
+  "EisConnector" when {
+
+    "retrieveUnderTaking is called" should {
+
+      "return an UnderTaking for a successful request" in {
```

Comment:
can we add test case like when the request json can't be parsed?

Reply:
Yeah, this is just the first cut of the test to get the basic pattern down. I will be adding further test cases as part of the work I'm doing in the next PR.

Reply:
👍